teamspeak_client: update pluginsdk

priegger opened the pull request:
## Description of changes

The old url was not reachable for me. This replaces it with the current version from teamspeak.com and updates the package for it.

In addition:
* move distribution permit email to a separate txt file
* reformat file using nixfmt

Fixes https://github.com/NixOS/nixpkgs/issues/344885

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Atemu commented on pkgs/applications/networking/instant-messengers/teamspeak/client.nix:
```diff
@@ -114,43 +161,45 @@ stdenv.mkDerivation rec {
       url = "https://www.teamspeak.com/en/privacy-and-terms/";
       free = false;
     };
-    maintainers = with maintainers; [ lhvwb lukegb atemu ];
+    maintainers = with maintainers; [
+      lhvwb
+      lukegb
+      atemu
+    ];
     mainProgram = "ts3client";
     platforms = [ "x86_64-linux" ];
   };
 }
 
 /*
-License issues:
-Date: Mon, 10 Dec 2007 19:55:16 -0500
-From: TeamSpeak Sales <sales@tritoncia.com>
-To: 'Marc Weber' <marco-oweber@gmx.de>
-Subject: RE: teamspeak on nix?
-
-Yes, that would be fine.  As long as you are not renting servers or selling
-TeamSpeak then you are more than welcome to distribute it.
-
-Thank you,
+  License issues:
+  Date: Mon, 10 Dec 2007 19:55:16 -0500
+  From: TeamSpeak Sales <sales@tritoncia.com>
+  To: 'Marc Weber' <marco-oweber@gmx.de>
+  Subject: RE: teamspeak on nix?
```
Could you move this to a separate .txt file and merely reference it here so that it doesn't get mangled by any future formatting?

Atemu reviewed (approved):
I don't like relying on IA for this as this effectively makes them effectively function as a CDN; all Nix TS3 users will hit this url.

Your git user is broken and/or its email not connected to your GH account.

I prefer to keep my packages not nixfmt'd because it's still in development and produces suboptimal results in some cases and the mass-format will happen anyways when it's ready. But in this case it doesn't look too bad.

priegger commented:
Thanks for the review, I can undo the formatting after the weekend.

For me the build was broken because oft the unreachable download and I saw IA used in other places. What do you recommend to do instead?

Atemu commented:
I don't have an alternative, hence the approval.

What I'm mainly concerned with is your broken git user. Please correct that and we're good.

You can keep the reformat but I'd like the permission email to be moved to a separate non-nix file.

Shawn8901 commented:
Can't we use that as source? https://github.com/teamspeak/ts3client-pluginsdk

It's the link on their website for the pluginsdk (it's quite new linked there)

priegger committed:
teamspeak_{client,server}: move distribution note

Move the distribution permit to a separate file to prevent mangling by
formatters.

priegger committed:
teamspeak_client: format with nixfmt

priegger committed:
teamspeak_client: update pluginsdk

The old url is not reachable anymore.

priegger commented on pkgs/applications/networking/instant-messengers/teamspeak/client.nix:
```diff
@@ -38,39 +83,36 @@ stdenv.mkDerivation rec {
-  pluginsdk = fetchurl {
-    url = "http://dl.4players.de/ts/client/pluginsdk/pluginsdk_3.1.1.1.zip";
-    sha256 = "1bywmdj54glzd0kffvr27r84n4dsd0pskkbmh59mllbxvj0qwy7f";
+  pluginsdk = fetchzip {
```
Was there a reason not to use fetchzip? With unzip (as before) the file is extracted into a directory named `ts_sdk_3.3.1` which makes it a little more difficult.

Atemu replied:
None. I always prefer the most specific fetcher too.

priegger commented on pkgs/applications/networking/instant-messengers/teamspeak/distribution-permit.txt:
```diff
@@ -0,0 +1,32 @@
+License issues:
```
Suggestions for better file names are welcome.

Atemu replied:
Sounds good to me.

priegger commented on pkgs/applications/networking/instant-messengers/teamspeak/server.nix:
```diff
@@ -59,43 +59,9 @@ in stdenv.mkDerivation rec {
 }
-
-/*
-License issues:
```
This was the same as in client.nix, so I dropped it and added the note above. But the license in `meta` is different, so for me it looks like the note is not necessary.

priegger commented:
Note: I only tested the build so far ad did not install/run the application yet.

```
NIXPKGS_ALLOW_UNFREE=1 nix build --impure .#pkgs.teamspeak_client
```

Atemu reviewed (approved):
Diff LGTM, will test execution in a bit.

Atemu commented:
@ofBorg eval

priegger commented:
Result of `nixpkgs-review pr 344608` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>teamspeak_client</li>
  </ul>
</details>
